Firestore/core/include/firebase/firestore/timestamp.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_TIMESTAMP_H_
+#define FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_TIMESTAMP_H_
+
+#if !defined(_STLPORT_VERSION)
+#include <chrono>
+#endif
+#include <cstdint>
+#include <ctime>
+
+namespace firebase {
+
+/**
+ * A Timestamp represents a point in time independent of any time zone or
+ * calendar, represented as seconds and fractions of seconds at nanosecond
+ * resolution in UTC Epoch time. It is encoded using the Proleptic Gregorian
+ * Calendar which extends the Gregorian calendar backwards to year one. It is
+ * encoded assuming all minutes are 60 seconds long, i.e. leap seconds are
+ * "smeared" so that no leap second table is needed for interpretation. Range is
+ * from 0001-01-01T00:00:00Z to 9999-12-31T23:59:59.999999999Z.
+ *
+ * @see
+ * https://github.com/google/protobuf/blob/master/src/google/protobuf/timestamp.proto
+ */
+class Timestamp {
+ public:
+  /**
+   * Creates a new timestamp representing the epoch (with seconds and
+   * nanoseconds set to 0).
+   */
+  Timestamp();
+
+  /**
+   * Creates a new timestamp.
+   *
+   * @param seconds The number of seconds of UTC time since Unix epoch
+   *     1970-01-01T00:00:00Z. Must be from 0001-01-01T00:00:00Z to
+   *     9999-12-31T23:59:59Z inclusive.
+   * @param nanoseconds The non-negative fractions of a second at nanosecond
+   *     resolution. Negative second values with fractions must still have
+   *     non-negative nanoseconds values that count forward in time. Must be
+   *     from 0 to 999,999,999 inclusive.
+   */
+  Timestamp(std::int64_t seconds, std::int32_t nanoseconds);
+
+  /**
+   * Creates a new timestamp with the current date.
+   *
+   * If <chrono> is available, precision will correspond to the precision of
+   * `std::chrono::system_clock`. Otherwise, the timestamp will have second
```

Comment:
I'm not sure which dialect of JavaDoc we use for C++ and, consequently, whether escaping code with backticks is valid.

Reply:
It's probably doxygen (but that's a guess; I really have no idea), in which case, what you've done looks correct.

https://www.stack.nl/~dimitri/doxygen/manual/markdown.html#md_codespan

Your `chrono` might cause problems though. Escaping the angle brackets with a backslash or surrounding it in backticks would probably help.

(But all that said, I wouldn't worry too much about the doxygen formatting just yet. Since we're not actually generating the docs yet, this whole discussion is likely premature.)